### PR TITLE
Story/vspc 48

### PR DESCRIPTION
**Pull request by prjadhav14**

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		IModule module = moduleManager.getModule(moduleId);
+		model.addAttribute("module", module);
+		List<ISequence> list=moduleManager.getModuleSequences(moduleId);
+		boolean flag=true;
+		for(ISequence sequence:list) {
+			if(sequence.getId().equals(sequenceId)) {
+				flag=false;
+				break;
+			}
+		}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java:36]: use the stream and filter API of collections here

Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java:36]: it's probalby even better to add a method to the sequence repo that gets a sequence by id and module.

File: vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	@Autowired
+	private IModuleManager moduleManager;
+
+	@Autowired
+	private ISequenceManager sequenceManager;
+
+	@RequestMapping(value = "/exhibit/module/{moduleId}/sequence/{sequenceId}")
+	public String sequence(Model model, @PathVariable("sequenceId") String sequenceId, @PathVariable("moduleId") String moduleId) {
+		IModule module = moduleManager.getModule(moduleId);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java:27]: what if the module doesn't exist?

File: vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSlideController.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	public String slide(Model model, @PathVariable("slideId") String slideId,@PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+		IModule module = moduleManager.getModule(moduleId);
+		model.addAttribute("module", module);
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
+		model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
+		model.addAttribute("slides", sequenceManager.getSequence(sequenceId).getSlides());
+		model.addAttribute("currentSequenceId", sequenceId);
+		model.addAttribute("currentSlide", slideId);
+		model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+		model.addAttribute("startSequenceId", module.getStartSequence().getId());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSlideController.java:40]: what if module or start sequence doesn't exist? or if slide is not part of the module?

Comment (prjadhav14) [reply vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSlideController.java:40]: These controllers are not part of story 48 and all the above comments are addressed in story 58. I will try to delete these files from this story.

**Pull request closed without merging**